### PR TITLE
Crudclass

### PR DESCRIPTION
**Pull request by Muhannadimad**

### CODE REVIEW

File: Models/Mercurycontext.cs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        // public MercuryContext(): base(){
+        //     Database.SetInitializer(new SchoolDBInitializer());
+        // }
+        //public DbSet<Student> Students { get; set; }
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        {
+            if (!optionsBuilder.IsConfigured)
+            {
+
+                optionsBuilder.UseMySql("server=localhost; database=MercuryMercury; user=root; pwd=123123;");
```
Comment (ZaidDireya) [inline Models/Mercurycontext.cs]: move this to configuration https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-3.1

File: Models/Mercurycontext.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+//     //     modelBuilder.Entity<Student>().HasKey(s => new { s.StudentId, s.FirstName });      
+
+//     //     //   modelBuilder.Entity<Student>()
+//     //     //         .Property(p => p.DateOfBirth)
+//     //     //         .HasColumnName("DoB")
+//     //     //         .HasColumnOrder(3)
+//     //     //         .HasColumnType("datetime2");
+//     }
+// }
```
Comment (ZaidDireya) [inline Models/Mercurycontext.cs]: remove commented code

File: Models/Mercurycontext.cs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+// public class Grade
+// {
+//     [Column(Order = 0)]
+//     public int Id { get; set; }
+//      [Column(Order = 2)]
+//     public string GradeName { get; set; }
+//      [Column(Order = 1)]
+//     public string Section { get; set; }
```
Comment (ZaidDireya) [inline Models/Mercurycontext.cs]: remove all commented code

File: Models/Order.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+        [Key]
+        public int Id { get; set; }
+    
+        public int ProductId { get; set; }
+      
+        public int UserId { get; set; }
+        public int Quantity { get; set; }
+
+    }
+}
```
Comment (ZaidDireya) [inline Models/Order.cs:13]: new line

File: Models/product.cs [hunk 5]
```diff
@@ -0,0 +1,18 @@
+using System;
+using System.ComponentModel.DataAnnotations;
+
+namespace Mercury.Models
+{
+    public class Product
+    {
+
```
Comment (ZaidDireya) [inline Models/product.cs]: add validation annotation https://docs.microsoft.com/en-us/aspnet/mvc/overview/getting-started/database-first-development/enhancing-data-validation

File: Program.cs [hunk 6]
```diff
@@ -1,57 +1,52 @@
...
+        // m.SaveChanges();
 
-        }
+
+
+     
+
+      var context = new MercuryContext();
+      
+      var x =   context.Product.Where(s => s.Id ==3).ToList();
```
Comment (ZaidDireya) [inline Program.cs]: extra space

File: Program.cs [hunk 7]
```diff
@@ -1,57 +1,52 @@
...
+
+          // add new entity :
+        // var p = new Product(){
+        //  Description = "bmw x5",
+        //  Price = 200000,
+        //  Quantity = 5
+        // };
+        // var m= new MercuryContext();
+        // m.Add(p);
+        // m.SaveChanges();
```
Comment (ZaidDireya) [inline Program.cs]: remove commented code

File: Program.cs (hunk 6 above)
Comment (ZaidDireya) [inline Program.cs]: why not using the repo

File: db-script/db.sql [hunk 8]
```diff
@@ -1,30 +1,27 @@
...
+  PRIMARY KEY (p_id)
+) CREATE TABLE order (
+  o_id INT NOT NULL AUTO_INCREMENT UNIQUE,
+  id INT NOT NULL,
+  p_id INT NOT NULL,
+  quantity INT NOT NULL,
+  PRIMARY KEY (o_id),
+  FOREIGN KEY (id) REFERENCES Persons(id),
+  FOREIGN KEY (p_id) REFERENCES Persons(p_id)
+)
```
Comment (ZaidDireya) [inline db-script/db.sql]: this file no longer needed

File: repo/ProductRepo.cs [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+//             arr = Read_All();
+//             for (int i = 0; i < arr.Length; i++)
+//                 Console.WriteLine(arr[i].p_id + " -- " + arr[i].p_price + " -- " + arr[i].p_quantity + " -- " + arr[i].p_description + " -- " + arr[i].created_at + " -- " + arr[i].updated_at);
+
+//         }
+//     }
+
+
+// }
+
```
Comment (ZaidDireya) [inline repo/ProductRepo.cs]: remove commented code, use the entity framework to do create, update, delete and get

File: repo/product.cs [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+        public string p_id = "";
+        public string p_price = "";
+        public string p_quantity = "";
+        public string p_description = "";
+        public string p_image = "";
+        public string created_at = "";
+        public string updated_at = "";
+
+    }
+    }
```
Comment (ZaidDireya) [inline repo/product.cs]: is this needed we already have a product model

Comment (Muhannadimad) [reply repo/product.cs]: i forget to delete it it is not a part of project

File: Program.cs (hunk 6 above)
Comment (Muhannadimad) [reply Program.cs]: i forget to delete the repo file it is not a part of project , the repo folder use ADO.NET to manipulation database ,but we want to use entity framework to manipulate database .

File: db-script/db.sql (hunk 8 above)
Comment (Muhannadimad) [reply db-script/db.sql]: should i create a new one ?
i think we should not because we have migration folder .

**Pull request closed without merging**